tfe/resource_tfe_variable.go

```diff
@@ -10,428 +10,709 @@ import (
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-		},
-		Schema: map[string]*schema.Schema{
-			"key": {
-				Type:     schema.TypeString,
-				Required: true,
+			"key": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the variable.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIf(
+						func(ctx context.Context, req planmodifier.StringRequest, resp *stringplanmodifier.RequiresReplaceIfFuncResponse) {
+							var stateSensitive types.Bool
+							diags := req.State.GetAttribute(ctx, path.Root("sensitive"), &stateSensitive)
+							if diags.HasError() {
+								resp.Diagnostics.Append(diags...)
+								return
+							}
+							if stateSensitive.ValueBool() && req.PlanValue.ValueString() != req.StateValue.ValueString() {
+								resp.RequiresReplace = true
+							}
+						},
+						"Force replacement if key changed and sensitive is true",
+						"Force replacement if key changed and sensitive is true",
+					),
+				},
 			},
-
-			"value": {
-				Type:      schema.TypeString,
-				Optional:  true,
-				Default:   "",
-				Sensitive: true,
+			"value": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+				Sensitive:   true,
+				Description: "Value of the variable",
 			},
-
-			"category": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
-				ValidateFunc: validation.StringInSlice(
-					[]string{
+			"category": schema.StringAttribute{
+				Required:    true,
+				Description: `Whether this is a Terraform or environment variable. Valid values are "terraform" or "env".`,
+				Validators: []validator.String{
+					stringvalidator.OneOf(
 						string(tfe.CategoryEnv),
 						string(tfe.CategoryTerraform),
-					},
-					false,
-				),
+					),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
 			},
-
-			"description": {
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
 				Optional: true,
-				Default:  "",
+				Computed: true,
+				Default:  stringdefault.StaticString(""),
 			},
-
-			"hcl": {
-				Type:     schema.TypeBool,
+			"hcl": schema.BoolAttribute{
 				Optional: true,
-				Default:  false,
+				Computed: true,
+				Default:  booldefault.StaticBool(false),
 			},
-
-			"sensitive": {
-				Type:     schema.TypeBool,
+			"sensitive": schema.BoolAttribute{
 				Optional: true,
-				Default:  false,
+				Computed: true,
+				Default:  booldefault.StaticBool(false),
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplaceIf(
+						func(ctx context.Context, req planmodifier.BoolRequest, resp *boolplanmodifier.RequiresReplaceIfFuncResponse) {
+							if req.StateValue.ValueBool() && !req.ConfigValue.ValueBool() {
+								resp.RequiresReplace = true
+							}
+						},
+						"Force replacement if sensitive argument changed from true to false.",
```

Comment:
The forced replacement is only when `sensitive` changes from `true to false`, what about when there is a reverse change, `false -> true`?

Reply:
My understanding is: _that's_ fine, and the API will let you do it without a blip. It's just that the API won't let you turn sensitive _off,_ because it's not willing to reveal that prior value to you again once it's been contaminated with the sensitive flag.